lib/package.rb

```diff
@@ -17,23 +17,35 @@ def self.dependencies
     @dependencies
   end
 
-  def self.depends_on (dependency = nil)
+  def self.depends_on (dependency = nil, pkgmanager = nil, pkgopt = nil)
```

Comment:
So how will this work?  Do you have an example depends_on line with all the options?

Reply:
`depends_on` will add dep to array if it is a `crew` package, otherwise install from other package manager if `pkgmanager` is provided:
```ruby
depends_on <dep> <pkgmanager> (optional) <options> (optional)
```

Reply:
Do you have a practical example?  Something with actual parameters?

Reply:
Umm...  like:
```ruby
depends_on 'pyOpenSSL', 'pip3', "--root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
```

Reply:
(If only `dep` was provided, dependency will be added to original array)

Reply:
> Umm... like:
> 
> ```ruby
> depends_on 'pyOpenSSL', 'pip3', "--root #{CREW_DEST_DIR} --prefix #{CREW_PREFIX}"
> ```

This will be `pip3 install pyOpenSSL --root /usr/local --prefix /usr/local` in command line

Reply:
Gotcha, so in this case the dependency will be included in the package metadata.  I'm not so sure this is the best approach, however.  The pyOpenSSL dependency might be needed with another package.  What I want to get away from is leaving a bunch of unused dependencies behind after a package is removed.  I guess it's 6 one way and half a dozen the other.  I do like your idea, however.  We have the `def self.remove` function we could use to tap into other package managers to uninstall as well.  This could let us create separate gem or python crew packages that would simply be a wrapper around those package managers.  These crew packages wouldn't need any metadata either.